clients/redshift/cast_test.go

```diff
@@ -13,52 +13,104 @@ import (
 
 func (r *RedshiftTestSuite) TestReplaceExceededValues() {
 	{
-		// Masked, reached the DDL limit
-		assert.Equal(r.T(), constants.ExceededValueMarker, replaceExceededValues(stringutil.Random(int(maxRedshiftLength)+1), typing.String))
-	}
-	{
-		// Masked, reached the string precision limit
-		stringKd := typing.KindDetails{
-			Kind:                    typing.String.Kind,
-			OptionalStringPrecision: typing.ToPtr(int32(3)),
+		// Irrelevant data type
+		{
+			// Integer
+			assert.Equal(r.T(), "123", replaceExceededValues("123", typing.Integer, false))
+		}
+		{
+			// Returns the full value since it's not a struct or string
+			// This is invalid and should not happen, but it's here to ensure we're only checking for structs and strings.
+			value := stringutil.Random(int(maxRedshiftLength + 1))
+			assert.Equal(r.T(), value, replaceExceededValues(value, typing.Integer, false))
 		}
-
-		assert.Equal(r.T(), constants.ExceededValueMarker, replaceExceededValues("hello", stringKd))
 	}
 	{
-		// Struct and masked
-		assert.Equal(r.T(), fmt.Sprintf(`{"key":"%s"}`, constants.ExceededValueMarker), replaceExceededValues(fmt.Sprintf(`{"foo": "%s"}`, stringutil.Random(int(maxRedshiftLength)+1)), typing.Struct))
+		// Exceeded
+		{
+			// String
+			{
+				// TruncateExceededValue = false
+				assert.Equal(r.T(), constants.ExceededValueMarker, replaceExceededValues(stringutil.Random(int(maxRedshiftLength)+1), typing.String, false))
+			}
+			{
+				// TruncateExceededValue = false, string precision specified
+				stringKd := typing.KindDetails{
+					Kind:                    typing.String.Kind,
+					OptionalStringPrecision: typing.ToPtr(int32(3)),
+				}
+
+				assert.Equal(r.T(), constants.ExceededValueMarker, replaceExceededValues("hello", stringKd, false))
+			}
+			{
+				// TruncateExceededValue = true
+				superLongString := stringutil.Random(int(maxRedshiftLength) + 1)
+				assert.Equal(r.T(), superLongString[:maxRedshiftLength], replaceExceededValues(superLongString, typing.String, true))
+			}
+			{
+				// TruncateExceededValue = true, string precision specified
+				stringKd := typing.KindDetails{
+					Kind:                    typing.String.Kind,
+					OptionalStringPrecision: typing.ToPtr(int32(3)),
+				}
+
+				assert.Equal(r.T(), "hel", replaceExceededValues("hello", stringKd, true))
+			}
+		}
+		{
+			// Struct and masked
+			assert.Equal(r.T(), fmt.Sprintf(`{"key":"%s"}`, constants.ExceededValueMarker), replaceExceededValues(fmt.Sprintf(`{"foo": "%s"}`, stringutil.Random(int(maxRedshiftLength)+1)), typing.Struct, false))
```

Comment:
Should we add a test case where a struct value gets truncated?

Reply:
> Should we add a test case where a struct value gets truncated?

They don't get truncated right now because it's unclear how we would actually truncate the struct (e.g. which key do we truncate and still maintain valid JSON?)